File: frontend/henn-laszlo/projects/artwork-data-access/src/lib/models/admin-artwork-details.ts
export interface AdminArtworkDetails {
  readonly id: string;
  readonly titleHu: string;
  readonly titleEn: string | null;
  readonly year: number | null;
  readonly techniqueHu: string | null;
  readonly techniqueEn: string | null;
  readonly widthCm: number | null;
  readonly heightCm: number | null;
  readonly descriptionHu: string | null;
  readonly descriptionEn: string | null;
  readonly hasImage: boolean;
  readonly isPublished: boolean;
  readonly isFeatured: boolean;
  readonly displayOrder: number;
}

File: backend/src/Modules/Modules.Artwork.Application/Artworks/GetAdminById/AdminArtworkDetails.cs
namespace Modules.Artwork.Application.Artworks.GetAdminById;

public sealed record AdminArtworkDetails(
    Guid Id,
    string TitleHu,
    string? TitleEn,
    int? Year,
    string? TechniqueHu,
    string? TechniqueEn,
    decimal? WidthCm,
    decimal? HeightCm,
    string? DescriptionHu,
    string? DescriptionEn,
    bool HasImage,
    bool IsPublished,
    bool IsFeatured,
    int DisplayOrder);